### ISSUE TITLE
NotSupportedException on DataTable.Load() on .NET Core 2.0

### ISSUE THREAD

**Issue by nulltoken** [mgravell/fast-member#46]
While playing with Dapper 1.50.4-alpha1-00070, I've encountered a small issue in making the TableValuedParameter feature work.

_Full disclosure:_ I'm creating the issue in this repository (however there are great chances this may be related to dotnet core itself), hoping someone can point me at a PEBKAC issue or a workaround. Wouldn't be the case, I'd very grateful if you could point me to a better place to create an issue, and I'll happily do it.

Below some material to repro the issue.

**.csproj:**
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="FastMember" Version="1.1.0" />
  </ItemGroup>

</Project>
```

**Programs.cs:**
```csharp
using System;
using System.Collections.Generic;
using System.Data;
using FastMember;

namespace ConsoleApp1
{
    class Program
    {
        static void Main(string[] args)
        {
            var locs = new List<Thing>();
            locs.Add(new Thing());
            locs.Add(new Thing());
            locs.Add(new Thing());

            InsertLocations(locs);
        }

        private static void InsertLocations(IList<Thing> validLocations)
        {
            using (DataTable table = new DataTable())
            using (var reader = ObjectReader.Create(validLocations, "Id", "Label", "Value", "CreatedAt"))
            {
                table.Load(reader); // This goes sideways here.
            }
        }
    }

    public class Thing
    {
        public Thing()
        {
            Id = Guid.NewGuid();
            Label = "TI";
            Value = 17.2m;
            CreatedAt = DateTimeOffset.UtcNow;
        }

        public Guid Id { get; set; }
        public string Label { get; set; }
        public decimal? Value { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
    }
}
```

When I run the program, `table.Load(reader)` throws a `NotSupportedException`.

![image](https://user-images.githubusercontent.com/92363/32137865-3f5b9df4-bc28-11e7-98ed-40b51d538a49.png)

I've tried to update `System.Data.Common` to 4.3.0, but with no luck.

Thoughts?

/cc @NickCraver (BTW, thanks for the amazing work/battle porting Dapper to dotnet core 2 😍)

Related (maybe...):
 - https://github.com/dotnet/corefx/issues/19748
 - https://github.com/dotnet/corefx/issues/24502

**Comment by trilm208** [comment]
.Net standard 2.0 same issue

**Comment by mgravell** [comment]
This is fixed in 1.2.0, deployed today
